test/foundry/Deposit.t.sol

```diff
@@ -0,0 +1,500 @@
+
+        // Prank as Alice to grant role
+        vm.startPrank(alice);
+
+        vault.grantRole(vault.DEPOSITOR(), bob);
+
+        // Prank as Bob for transactions
+        vm.startPrank(bob);
+
+        vm.expectRevert(abi.encodeWithSelector(ZeroAssetsAmount.selector));
+        vault.deposit(assets, bob, minShareRatio, receiptInformation);
+
+        vm.stopPrank();
+    }
+
+    function testZeroReceiver(
+        uint256 fuzzedKeyAlice,
+        uint256 fuzzedKeyBob,
+        string memory assetName,
+        string memory assetSymbol,
+        uint256 assets,
+        uint256 minShareRatio,
+        bytes memory receiptInformation
+    ) external {
+        minShareRatio = bound(minShareRatio, 0, 1e18);
+
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        address alice = vm.addr((fuzzedKeyAlice % (SECP256K1_ORDER - 1)) + 1);
+
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        address bob = vm.addr((fuzzedKeyBob % (SECP256K1_ORDER - 1)) + 1);
+
+        // Assume that assets is not 0
+        vm.assume(assets != 0);
+
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        // Prank as Alice to set role
+        vm.startPrank(alice);
+
+        vault.grantRole(vault.DEPOSITOR(), bob);
+
+        // Prank as Bob for transaction
+        vm.startPrank(bob);
+
+        vm.expectRevert(abi.encodeWithSelector(ZeroReceiver.selector));
+        vault.deposit(assets, address(0), minShareRatio, receiptInformation);
+
+        vm.stopPrank();
+    }
+
+    /// Test deposit to someone else reverts if system not certified
+    function testDepositToSomeoneElseExpiredCertification(
+        uint256 fuzzedKeyAlice,
+        string memory assetName,
+        string memory assetSymbol,
+        uint256 fuzzedKeyBob,
+        uint256 assets,
+        uint256 minShareRatio,
+        bytes memory receiptInformation
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        address alice = vm.addr((fuzzedKeyAlice % (SECP256K1_ORDER - 1)) + 1);
+
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        address bob = vm.addr((fuzzedKeyBob % (SECP256K1_ORDER - 1)) + 1);
+
+        minShareRatio = bound(minShareRatio, 0, 1e18);
+        // Assume that assets is not 0
+        vm.assume(assets != 0);
+
+        vm.assume(alice != bob);
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        // Prank as Alice to set role
+        vm.startPrank(alice);
+        vault.grantRole(vault.DEPOSITOR(), bob);
+
+        // Prank as bob for transaction
+        vm.startPrank(bob);
+
+        vm.expectRevert(abi.encodeWithSelector(CertificationExpired.selector, address(0), alice, 0, 1));
```

Comment:
fuzz 0 and 1

Reply:
This is a case to check depositing while system is not certified. so the certifiedUntil in CertificationExpired error is 0. I fuzzed timestamp and added new test to check depositing while certification is expired